test/os_linux_generic.py

```diff
@@ -29,433 +29,321 @@ class LinuxPortTestCase(unittest.TestCase):
+            "/dev/disk/by-id/ata-HDS728080PLA380_40Y9028LEN_PFDB32S7S44XLM-part5",
+            "/dev/disk/by-id/ata-TSSTcorpDVD-ROM_TS-H352C",
+            "/dev/disk/by-id/usb-MBED_FDi_sk_A000000001-0:0",
+            "/dev/disk/by-id/usb-MBED_microcontroller_0240020152986E5EAF6693E6-0:0",
+            "/dev/disk/by-id/wwn-0x5000cca30ccffb77",
+            "/dev/disk/by-id/wwn-0x5000cca30ccffb77-part1",
+            "/dev/disk/by-id/wwn-0x5000cca30ccffb77-part2",
+            "/dev/disk/by-id/wwn-0x5000cca30ccffb77-part5",
+        ],
+        "/dev/serial/by-id": [
+            "/dev/serial/by-id/usb-MBED_MBED_CMSIS-DAP_0240020152986E5EAF6693E6-if01",
+            "/dev/serial/by-id/usb-MBED_MBED_CMSIS-DAP_A000000001-if01",
+        ]
+    }
+
+    link_dict_1 = {
+        "/dev/disk/by-id/ata-HDS728080PLA380_40Y9028LEN_PFDB32S7S44XLM": "../../sda",
+        "/dev/disk/by-id/ata-HDS728080PLA380_40Y9028LEN_PFDB32S7S44XLM-part1": "../../sda1",
+        "/dev/disk/by-id/ata-HDS728080PLA380_40Y9028LEN_PFDB32S7S44XLM-part2": "../../sda2",
+        "/dev/disk/by-id/ata-HDS728080PLA380_40Y9028LEN_PFDB32S7S44XLM-part5": "../../sda5",
+        "/dev/disk/by-id/ata-TSSTcorpDVD-ROM_TS-H352C": "../../sr0",
+        "/dev/disk/by-id/usb-MBED_FDi_sk_A000000001-0:0": "../../sdc",
+        "/dev/disk/by-id/usb-MBED_microcontroller_0240020152986E5EAF6693E6-0:0": "../../sdb",
+        "/dev/disk/by-id/wwn-0x5000cca30ccffb77": "../../sda",
+        "/dev/disk/by-id/wwn-0x5000cca30ccffb77-part1": "../../sda1",
+        "/dev/disk/by-id/wwn-0x5000cca30ccffb77-part2": "../../sda2",
+        "/dev/disk/by-id/wwn-0x5000cca30ccffb77-part5": "../../sda5",
+        "/dev/serial/by-id/usb-MBED_MBED_CMSIS-DAP_0240020152986E5EAF6693E6-if01": "../../ttyACM1",
+        "/dev/serial/by-id/usb-MBED_MBED_CMSIS-DAP_A000000001-if01": "../../ttyACM0"
+    }
+
+    mount_list_1 = [
+        b"/dev/sdb on /media/usb0 type vfat (rw,noexec,nodev,sync,noatime,nodiratime,gid=1000,uid=1000,dmask=000,fmask=000)",
+        b"/dev/sdc on /media/usb1 type vfat (rw,noexec,nodev,sync,noatime,nodiratime,gid=1000,uid=1000,dmask=000,fmask=000)"
+    ]
     def test_get_detected_1_k64f(self):
-        # get_detected(self, tids, disk_list, serial_list, mount_list)
-
-        mbed_det = self.linux_generic.get_detected(self.tids,
-            self.disk_list_1,
-            self.serial_list_1,
-            self.mount_list_1)
-
-        self.assertEqual(1, len(mbed_det))
-        self.assertIn([
-            "FRDM_K64F",
-            "sdb",
-            "/media/usb0",
-            "/dev/ttyACM1",
-            "usb-MBED_microcontroller_0240020152986E5EAF6693E6-0:0 -> ../../sdb"
-          ],
+        mbed_det = self.find_candidates_with_patch(
+            self.mount_list_1, self.link_dict_1, self.listdir_dict_1)
+        self.assertIn({
+            'mount_point': "/media/usb0",
+            'serial_port': "/dev/ttyACM1",
+            'target_id': "0240020152986E5EAF6693E6"
```

Comment:
Nit: standardizing on `"` or `'` would be nice for consistency, not a hard requirement for this PR though.